libsolidity/codegen/YulUtilFunctions.cpp

```diff
@@ -3645,16 +3649,20 @@ string YulUtilFunctions::revertReasonIfDebug(string const& _message)
 	return revertReasonIfDebug(m_revertStrings, _message);
 }
 
-string YulUtilFunctions::panicFunction()
+string YulUtilFunctions::panicFunction(util::PanicCode _code)
 {
-	string functionName = "panic_error";
+	string functionName = "panic_error_" + toCompactHexWithPrefix(uint64_t(_code));
 	return m_functionCollector.createFunction(functionName, [&]() {
 		return Whiskers(R"(
 			function <functionName>() {
-				invalid()
+				mstore(0, <selector>)
+				mstore(4, <code>)
+				revert(0, 0x24)
```

Comment:
Our selector is 4 byte and code is 2 byte? Can't we pack everything in 32 bytes?

Reply:
We want to use ABI encoding. Furthermore, there is a missing feature in the RPC interface to the node implementations in that they do not return the success code of a transaction. Computing the length of the return value mod 32 is an indication of whether it succeeded or not.